src/Compilers/Core/Portable/Compilation/OperationExtensions.cs

```diff
@@ -16,28 +16,27 @@ public static class OperationExtensions
     {
         public static IEnumerable<IOperation> Descendants(this IOperation operation)
         {
-            var searchable = operation as IOperationSearchable;
-            if (searchable != null)
-            {
-                return searchable.Descendants();
-            }
-            else
+            if (operation == null)
             {
                 return SpecializedCollections.EmptyEnumerable<IOperation>();
             }
+            var list = new List<IOperation>();
+            var collector = new OperationCollector(list);
+            collector.Visit(operation);
+            list.RemoveAt(0);
+            return list;
         }
 
         public static IEnumerable<IOperation> DescendantsAndSelf(this IOperation operation)
         {
-            var searchable = operation as IOperationSearchable;
```

Comment:
Probably. @mattwar created the IOperationSearchable mechanism in order to implement Descendants***. I'm not aware of it having other utility.